### ISSUE TITLE
ESP32 lwmqtt connection denied

### ISSUE THREAD

**Issue by kishoreinme** [GoogleCloudPlatform/google-cloud-iot-arduino#83]
Hello

I am trying out the v1.1.0 esp32-lwmqtt sample with the following board with connection denied on mqtt_connect

https://www.cnx-software.com/2017/10/13/this-ttgo-board-combines-esp32-lora-radio-and-oled-display-for-just-10/
I  rechecked my device, registry, project and private key ciotc config. Infact I follow the same commands (even the names) in readme to tryout things first time by keeping  it simple. 

On the root certifcate part, the  openssl command  in the instructions of ciotc returns a certifcate chain with two certificates in the gcould shell  and I am not sure which one to copy. I just tried with the first certificate.

And also it says to copy the content to root.cert. I do not understand where the root.cert should be placed (or flashed into esp32 with some tool?!?!)

Any help is appreciated. Thanks

**Comment by jesuspisa** [comment]
Hi @kishoreinme I am trying to connect an sparkfun esp32 thing to google cloud iot too. I have followed the steps in this page: https://medium.com/@ludovic.emo/how-to-send-esp32-telemetry-to-google-cloud-iot-core-caf1a952020d

I saved all the certificates created by following the previous steps. I saved them in the directory where I saved my example Arduino Ide Progam. In my case C:\Documents\Arduino\my Esp32\Google Cloud Iot

Look at this picture: 

![image](https://user-images.githubusercontent.com/37476785/52533385-58541c00-2d33-11e9-848b-525159d9a50d.png)


Then I uploaded at google cloud iot core the iot-sample-device-ec_public.pem to the device created at google cloud iot. 

![image](https://user-images.githubusercontent.com/37476785/52533381-4ecab400-2d33-11e9-8dc5-4e1598278b9d.png)

Even if I did these steps, once I compiled my Arduino Ide Program I have this error: 

ssl_client.cpp: In function 'int start_ssl_client(sslclient_context*, const char*, uint32_t, const char*, const char*, const char*, const char*, const char*)':

C:\Users\Jesus\Documents\Arduino\libraries\WiFiClientSecure\src\ssl_client.cpp:144:67: error: 'mbedtls_ssl_conf_psk' was not declared in this scope

                  (const unsigned char *)pskIdent, strlen(pskIdent));


Do you have any compilation errors? Or Does it work perfectly to you? 
I compiled the Google Cloud Iot example appeared in the first link I shared: https://medium.com/@ludovic.emo/how-to-send-esp32-telemetry-to-google-cloud-iot-core-caf1a952020d

¡Thank you!

**Comment by kishoreinme** [comment]
Hello @jesuspisa 
I did not face any compilation error with wifisecureclient. I just use the latest default libs.  And in my case I dont understand who will upload the keys to the board. 
Meanwhile ,  my use case  is fairly simple  with very few number of devices which are always powered, with enough bandwidth and did not need a full IOT suite with MQTT  like the GCP IOT core. I just wanted to try it out.
So I have moved to firebase with HTTP.
